app/controllers/APIController.scala

```diff
@@ -22,9 +26,64 @@ class APIController @Inject() (
     status.asJson.spaces2
   }
 
+  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
+
+    Console.println(request.body.asText)
+    val parseResult: Option[ProductFromJson] = {
+      request.body.asText match {
+        case None =>
+          BadRequest("Request was invalid, nothing in request body")
```

Comment:
This bad request is superfluous and not used. the return value here is the `None` below it